### DISCUSSION TITLE
How to call onSignIn method when we are using ConnectKitButton?

### DISCUSSION THREAD

**Discussion by itsanishjain** [family/connectkit#113] [Q&A] [answered]
I am using singIn with Ethereum and when I am using ConnectKitButton 
I get this screen and when I click signIn here I need to call some backend function to save User's wallet address into my database

![Screenshot from 2023-02-09 14-52-58](https://user-images.githubusercontent.com/43033153/217771866-558e893b-a1ef-422c-a82a-a9466a98df8a.png)

but there is no onSignIn method similar to when I am using

```
<SIWEButton
        showSignOutButton
        onSignIn={() => {
          console.log("Sign");
          // call /api/user
        }}
      />
 ```
 
 Is there any way to achieve the same with that with ConnectKitButton flow.

**Comment by breath-imager** [comment]
I came here to post the exact same thing. Or at least a way to disable this signin and force people to sign in using SIWEbutton.

**Reply by itsanishjain** [reply]
@breath-imager 
Well if you go to this page
https://docs.family.co/connectkit/api-reference#additional-components-chain-icon

and scroll down to **SIWE Button** they did something similar but I didn't find a code for it.

but that parameter is I guess really something we needed.

 I am using this code to get SigneIn status and once it's true I will call the API in useEffect but it's not very clean.

``` const { signedIn, address } = useSIWE();```

**Comment by itsanishjain** [comment]
@lochie Hey can you provide any recommendations on this

**Comment by lochie** [comment]
`onSignIn` and `onSignOut` coming in [this PR](https://github.com/family/connectkit/pull/69/)!
This will be usable on the SIWE Provider for a top-level response, and/or individually on each `useSIWE` Hook (and the `SIWEButton`) which will respond when that that hook in particular activates the event.

**Comment by lochie** [comment, marked as answer]
This feature is now available in ConnectKit 1.2.0!

Check out the release notes
https://github.com/family/connectkit/releases/tag/v1.2.0

**Reply by breath-imager** [reply]
Amazing, thanks!
